### PR TITLE
VULCAN-370: Delete or rename project

### PR DESCRIPTION
**Pull request by freddyfeelgood**
Added necessary functionality to handle deleting and renaming a project.
Closes: #370

### CODE REVIEW

File: app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -138,5 +138,6 @@ def new_project_params
 
   def project_params
     params.require(:project).permit(project_metadata_attributes: { data: {} })
+    params.require(:project).permit(:name)
```
Comment (rbclark) [inline app/controllers/projects_controller.rb:141]: I'm pretty sure removing this breaks project metadata.

File: app/javascript/components/projects/RenameProjectModal.vue [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      this.$refs["renameProjectModal"].hide();
+      let payload = { project: { name: this["name"] } };
+      axios
+        .put(`/projects/${this.project.id}`, payload)
+        .then(this.renameProjectSuccess)
+        .catch(this.alertOrNotifyResponse);
+    },
+    renameProjectSuccess: function (response) {
+      this.alertOrNotifyResponse(response);
+      location.reload();
```
Comment (rbclark) [inline app/javascript/components/projects/RenameProjectModal.vue:64]: It shouldn't be necessary to reload the whole page to rename the project.

File: app/javascript/components/projects/ProjectsTable.vue [hunk 3]
```diff
@@ -44,6 +44,28 @@
       <template #cell(updated_at)="data">
         {{ friendlyDateTime(data.item.updated_at) }}
       </template>
+
+      <template #cell(actions)="data">
+        <RenameProjectModal
+          :project="data.item"
```
Comment (timwongj) [inline app/javascript/components/projects/ProjectsTable.vue:50]: @rlakey should non-admin users be able to rename a project? currently the back end code only allows admins to rename or remove a project but here a regular user is shown the rename button

Comment (rlakey) [reply app/javascript/components/projects/ProjectsTable.vue:50]: A Vulcan admin or a project admin?

Comment (timwongj) [reply app/javascript/components/projects/ProjectsTable.vue:50]: vulcan admin

Comment (rlakey) [reply app/javascript/components/projects/ProjectsTable.vue:50]: I think we're fine if either a vulcan admin or a project admin can rename.

Comment (rlakey) [reply app/javascript/components/projects/ProjectsTable.vue:50]: If you want to hide the button for users that aren't admins that's fine too.

File: app/javascript/components/projects/ProjectsTable.vue [hunk 4]
```diff
@@ -110,8 +146,25 @@ export default {
...
       return "/projects/" + project.id;
     },
+    destroyAction: function (project) {
+      return `/projects/${project.id}`;
+    },
+    getLabel: function (project) {
+      return (
+        "Are you sure you want to completely remove project " +
+        project.name +
+        " and all of its related data?"
```
Comment (xsseve) [inline app/javascript/components/projects/ProjectsTable.vue:156]:
```suggestion
        `Are you sure you want to completely remove project ${project.name} and all of its related data?`
```

File: app/javascript/components/projects/RenameProjectModal.vue [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    },
+    renameProjectSuccess: function (response) {
+      this.alertOrNotifyResponse(response);
+      this.$emit("projectRenamed");
+    },
+  },
+};
+</script>
+
+<style scoped></style>
```
Comment (xsseve) [inline app/javascript/components/projects/RenameProjectModal.vue:70]:
```suggestion
```

File: app/javascript/components/projects/Projects.vue [hunk 6]
```diff
@@ -16,6 +21,26 @@ export default {
...
+  },
+  data: function () {
+    return {
+      projectlist: this.projects,
+    };
+  },
+  methods: {
+    refreshProjects: function () {
+      axios
+        .get(`/projects`)
```
Comment (xsseve) [inline app/javascript/components/projects/Projects.vue:38]:
```suggestion
        .get('/projects')
```

File: app/javascript/components/projects/ProjectsTable.vue [hunk 7]
```diff
@@ -110,8 +146,25 @@ export default {
     getProjectAction: function (project) {
       return "/projects/" + project.id;
     },
+    destroyAction: function (project) {
+      return `/projects/${project.id}`;
```
Comment (xsseve) [inline app/javascript/components/projects/ProjectsTable.vue:150]: Map this string type (``) to other functions

**Review by xsseve — CHANGES_REQUESTED** [review]

File: app/javascript/components/projects/ProjectsTable.vue (hunk 7 above)
Comment (timwongj) [reply app/javascript/components/projects/ProjectsTable.vue:150]: https://github.com/mitre/vulcan/pull/386/commits/56dffa21e82963ad5974b300c6f5a7831aebc777

File: app/javascript/components/projects/Projects.vue (hunk 6 above)
Comment (timwongj) [reply app/javascript/components/projects/Projects.vue]: https://github.com/mitre/vulcan/pull/386/commits/56dffa21e82963ad5974b300c6f5a7831aebc777

File: app/javascript/components/projects/RenameProjectModal.vue (hunk 5 above)
Comment (timwongj) [reply app/javascript/components/projects/RenameProjectModal.vue:70]: this is left here in other files even if it's empty

File: app/javascript/components/projects/ProjectsTable.vue (hunk 4 above)
Comment (timwongj) [reply app/javascript/components/projects/ProjectsTable.vue]: https://github.com/mitre/vulcan/pull/386/commits/56dffa21e82963ad5974b300c6f5a7831aebc777

**Comment by timwongj** [comment]
hide rename from non-project admin users

**Comment by freddyfeelgood** [comment]
Tested renaming a project, deleting a project as vulcan admin, looks good.  Logged in as a user with project "reviewer" role, buttons not available.  Looks good to me.

**Comment by timwongj** [comment]
> Tested renaming a project, deleting a project as vulcan admin, looks good. Logged in as a user with project "reviewer" role, buttons not available. Looks good to me.

Thanks for taking a look!

**Pull request merged**